### PR TITLE
added support to launch and install ELK docker image

### PR DESCRIPTION
**Pull request by lbrack**
This is work ing progress!﻿

### CODE REVIEW

File: docs/usage.rst [hunk 1]
```diff
@@ -152,20 +152,146 @@ To obtain a complete report for a given result, you can use the built-in method
...
 
 API
 ***
 
-There are three sub-modules under :py:mod:`testspace_colab`:
+There are several sub-modules under :py:mod:`testspace_colab`:
 
     * :py:mod:`testspace_colab.lib` which provides the
-      :py:class:`API <testspace_colab.lib.API>` class
+      :py:class:`API <testspace_colab.lib.API>` class ro interface
```
Comment (zy1o) [inline docs/usage.rst:243]: gotcha! typo! should be "class to interface"

File: src/testspace_colab/elk.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        try:
+            container = self._docker.containers.get(self._container_name)
+            if container.status == "running" and not self._elastic_search:
+                self._elastic_search = elasticsearch.Elasticsearch()
+                self._cluster_client = elasticsearch.client.ClusterClient(
+                    self._elastic_search
+                )
+
+        except (docker.errors.NotFound, docker.errors.APIError):
+            pass
```
Comment (zy1o) [inline src/testspace_colab/elk.py:74]: Maybe a good idea to log what went wrong here? Just to make debugging easier when things go south.

File: src/testspace_colab/elk.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        if not self.container:
+            logger.info(f"Starting ELK Container {self._container_name}")
+            self._docker.containers.run(
+                image=f"sebp/elk:{self._elk_tag}",
+                name=self._container_name,
+                ports={5601: 5601, 9200: 9200, 5044: 5044},
+                tty=True,
+                detach=True,
+            )
+            time.sleep(5)
```
Comment (zy1o) [inline src/testspace_colab/elk.py:174]: Why magic 5 seconds? If there's a reason, add a comment.

File: src/testspace_colab/elk.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+        elif self.container.status != "running":
+            logger.info(
+                f"Starting existing container - current state {self.container.status}"
+            )
+            self.container.start()
+            time.sleep(5)
```
Comment (zy1o) [inline src/testspace_colab/elk.py:180]: Again, I don't like the hardcoded 5.

File: src/testspace_colab/lib.py [hunk 5]
```diff
@@ -29,13 +29,13 @@ def __init__(self, token=None, url=None, project=None, space=None):
...
-            pathlib.Path(os.path.expanduser("~")) / ".config" / "testspace" / "config"
-        )
-        if "TS_COLAB_CONFIG" in os.environ:
-            config_path = pathlib.Path(os.environ["TS_COLAB_CONFIG"])
-            assert config_path.is_file(), f"{config_path} not found"
-            click.secho(f"using TS_COLAB_CONFIG={config_path}")
+        config_dir = pathlib.Path(os.path.expanduser("~")) / ".config" / "testspace"
+        if "TS_COLAB_CONFIG_DIR" in os.environ:
+            config_dir = pathlib.Path(os.environ["TS_COLAB_CONFIG_DIR"])
+            assert config_dir.is_dir(), f"{config_dir} dir not found"
```
Comment (zy1o) [inline src/testspace_colab/lib.py:35]: I think that you're better off using another exception. Something like  `if not config_dir.is_dir(): raise OSError(f"{config_dir} is not a directory")` feels better.
First, assert statements in my mind feel like it's a test. It is implemented as `if PY_DEBUG: raise AssertionError`, you're safer using other exceptions.

**Review by zy1o — COMMENTED** [review]
Overall I picked up a few minor things that I'd probably change.

File: src/testspace_colab/elk.py (hunk 3 above)
Comment (lbrack) [reply src/testspace_colab/elk.py:174]: I opened an issue to remind me

File: src/testspace_colab/elk.py (hunk 2 above)
Comment (lbrack) [reply src/testspace_colab/elk.py:74]: see #4

File: src/testspace_colab/elk.py (hunk 3 above)
Comment (lbrack) [reply src/testspace_colab/elk.py:174]: Really None. Just a gut feeling

File: src/testspace_colab/elk.py (hunk 4 above)
Comment (lbrack) [reply src/testspace_colab/elk.py:180]: Well here is the thing. We know the ELK stack takes a while to initialize. So there is no point in bombarding it with requests to see if it is up and running. By the way, both "sleeps" are for the same purpose. In one case, it is to bootstrap a container from an image whereas in the other it is to simply start an existing container. 

By the way, when container.start() return, the container is in running state already. However, there is the case where the image is not yet on your machine.

Comment (lbrack) [reply src/testspace_colab/elk.py:180]: Actually let me take that back ... this is all done in the docker.model.container.run so this is a non issue

**Pull request merged**